### DISCUSSION TITLE
Display list of static files on static site

### DISCUSSION THREAD

**Discussion by theS1LV3R** [nuxt/nuxt#9496] [Questions (Nuxt 2)] [answered]
I'm currently in the process of rewriting my website from Jekyll to Nuxt, and its working mostly great, but I've stumbled upon a problem: In my current website I have some video files that I'm listing on a page, but I don't know how I'd do that in Nuxt/Vue.

```html
<ul>
{%- for file in site.static_files -%} <!-- List static files -->
  {%- if file.path contains "/p/" -%} <!-- Check if it's in the correct folder (where i store the videos) -->
    <li><a href="{{file.path}}">{{file.name}}</a></li> <!-- And then link to the file -->
  {%- endif %}
{%- endfor -%}
</ul>
```
Here is an example of the output as of me writing this:
![Example of output](https://user-images.githubusercontent.com/32072697/123879713-b6541f00-d941-11eb-8883-5b22650defe8.png)

I've thought of a couple ways to possibly do this:
- Using the content folder and fetching from there
- Using webpack somehow?

This is a static page that will be deployed to Netlify, so no server-side stuff can be used.

How would I create this in Nuxt?

---

Current source can be found here, don't have a repo for the nuxt project yet: https://github.com/s1lv3r-website/landing

**Comment by kissu** [comment, marked as answer]
This should be good enough and you'll probably benefit from some Webpack performance optimizations (at some point).
![image](https://user-images.githubusercontent.com/5133074/124104424-8fd9d500-da62-11eb-95fe-1ba8f4408b52.png)

**Reply by theS1LV3R** [reply]
That is with hardcoding the file paths and embedding them, but is there any way to just list the files? See like the example screenshot

**Reply by kissu** [reply]
Oh sorry if I did not get it right from the beginning!
If you want to list the files available in a specific directory, you could use the following
```html
<script>
export default {
  mounted() {
    const videos = require.context('@/assets/videos', true, /^.*\.mp4$/)
    console.log(videos.keys())
  },
}
</script>
```

As shown here: https://stackoverflow.com/a/58091294/8816585

**Reply by theS1LV3R** [reply]
How would I use this while using typescript? I just get an error:
```error
Property 'context' does not exist on type 'NodeRequire'. Vetur(2339)
```

Even after installing `@types/webpack-env` and adding the types to tsconfig.json as suggested by multiple searches it still doesnt work

**Reply by kissu** [reply]
I don't know TS, cannot help sorry. 😅

**Reply by theS1LV3R** [reply]
Alright, thanks for the help anyways! I think I know what I'll need to do now :smile:

**Reply by kissu** [reply]
Yeah, need to learn TS too at some point. Glad it helped. 🙏🏼
